Description: Find global headers
 Have hyprpm find global headers installed with the package instead of requiring
 a separate clone.
Author: Constantin Piber <cp.piber@gmail.com>

Index: source/hyprpm/src/core/PluginManager.cpp
===================================================================
--- source.orig/hyprpm/src/core/PluginManager.cpp
+++ source/hyprpm/src/core/PluginManager.cpp
@@ -391,6 +391,9 @@ bool CPluginManager::removePluginRepo(co
 eHeadersErrors CPluginManager::headersValid() {
     const auto HLVER = getHyprlandVersion(false);
 
+    if (std::filesystem::exists("/usr/share/pkgconfig/hyprland.pc"))
+        return HEADERS_OK;
+
     if (!std::filesystem::exists(DataState::getHeadersPath() + "/share/pkgconfig/hyprland.pc"))
         return HEADERS_MISSING;
 
@@ -463,6 +466,17 @@ bool CPluginManager::updateHeaders(bool
         std::filesystem::permissions(getTempRoot(), std::filesystem::perms::owner_all, std::filesystem::perm_options::replace);
     }
 
+    if (headersValid() != HEADERS_OK) {
+        std::println("\n{}", failureString("Headers not okay. File a bug with https://github.com/cpiber/hyprland-ppa/ (not Hyprland)!"));
+        return false;
+    }
+
+    auto GLOBALSTATE                = DataState::getGlobalState();
+    GLOBALSTATE.headersHashCompiled = HLVER.hash;
+    DataState::updateGlobalState(GLOBALSTATE);
+    std::println("\n{}", successString("Headers up to date."));
+    return true;
+
     if (!force && headersValid() == HEADERS_OK) {
         std::println("\n{}", successString("Headers up to date."));
         return true;
